src/snowflake/cli/api/project/schemas/native_app/path_mapping.py

```diff
@@ -1,10 +1,22 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import Any, Dict, List, Optional, Union
 
+from pydantic import Field
 from snowflake.cli.api.project.schemas.updatable_model import UpdatableModel
 
 
+class AnnotationProcessor(UpdatableModel):
+    name: str = Field(
+        title="Name of the processor to be invoked to discover the annotated code."
+    )
+    properties: Optional[Dict[str, Any]] = Field(
+        title="Properties to aid the invocation of the processor on the annotated code. If none are provided, then the Snowflake CLI will try to get the current environment information.",
```

Comment:
Ditto, this should be generic. Something like "Set of key-value pairs used to configure the output of the processor. Consult a specific processor's documentation for more details on supported properties."